src/map.cpp

```diff
@@ -544,19 +549,24 @@ vehicle *map::move_vehicle( vehicle &veh, const tripoint &dp, const tileray &fac
             if( coll.type == veh_coll_veh ) {
                 continue;
             }
-            if( static_cast<size_t>( coll.part ) > veh.parts.size() ||
-                veh.parts[coll.part].removed ) {
+            if( !veh.valid_part( coll.part, true ) ||
+                veh.part( coll.part ).removed ) {
                 continue;
             }
+            int part_num = coll.part;
+            if( veh.fake_part( part_num ) ) {
+                part_num = veh.get_fake_parent( part_num );
+            }
```

Comment:
These four lines are repeated several times. I suggest you move them into a function, so this can be shortened to
```suggestion
            const int part_num = get_non_fake_part( coll.part );
```
(Which either returns the input part index, or the fake parent.)